web-common/src/features/sources/modal/FileDrop.svelte

```diff
@@ -36,7 +34,7 @@
 
     const uploadedFiles = uploadTableFiles(
       Array.from(files),
-      [$sourceNames?.data ?? [], $modelNames?.data ?? []],
+      $allNames.data ?? [],
       $runtime.instanceId,
     );
     for await (const { tableName, filePath } of uploadedFiles) {
```

Comment:
**Security Issue**: The removal of specific file name selectors and replacement with 'useAllFileNames' increases the risk of processing unauthorized files, as it broadens the scope of accessible files without apparent restrictions or validations. <br> **Fix**: Reinstate specific file name selectors with added security checks to ensure only authorized files are processed. Implement a validation layer to filter out unauthorized access based on user roles and permissions. <br> **Code Suggestion**: 
 ```
 -      $allNames.data ?? [],
 +      $allNames.data.filter(name => userIsAuthorizedToAccessFile(name)) ?? [],
 ```